tests/testthat/test-format-report-external.R

```diff
@@ -0,0 +1,69 @@
+# These are slimmed down variants of some tests from test-format-report.R,
+# focused on checking if the formatting functions are compatible with the
+# objects read in from external results.
+
+results_dir <- local_create_external_results()
+
+describe("formatting functions with external scores", {
+  it("format_scores_for_render", {
+    pkg_scores <- build_pkg_scores(results_dir)
+    formatted <- format_scores_for_render(pkg_scores)[["formatted"]]
+
+    overall_scores <- formatted[["overall_scores"]]
+    expect_identical(
+      names(overall_scores),
+      c("category", "category_score", "risk")
+    )
+    expect_identical(
+      overall_scores[["category"]],
+      c(METRIC_CATEGORIES, "overall")
+    )
+
+    expect_identical(
+      purrr::map(formatted[["category_scores"]], "criteria"),
+      list(
+        testing = TESTING_METRICS,
+        documentation = c("has_website", "has_news"),
+        maintenance = MAINTENANCE_METRICS,
+        transparency = TRANSPARENCY_METRICS
+      )
+    )
+  })
+
+  it("format_testing_scores", {
+    pkg_scores <- build_pkg_scores(results_dir)
+    flex_df <- format_testing_scores(format_scores_for_render(pkg_scores))
+    res <- flex_df[["body"]][["dataset"]]
+    expect_identical(res[["Result"]], c("Passing", "91.54%"))
+  })
```

Comment:
What was the rational behind including a `format_testing_scores()` test, but not one for `format_package_details()`?

 - Im not sure you need either unless you're also testing the interactivity with `build_pkg_scores()`/`format_scores_for_render()`.

Reply:
My reason to include a `format_testing_scores` test (versus not or instead including a test for `format_package_details` or `format_overall_scores`) was to check an internal vs external difference.  Due to the `map_answer` change in this PR, you'd expect to see "Passing" rather than "Passing (score: ...)".

More generally, I didn't intend for this file to duplicate everything in `test-format-report.R` but wanted it to send the result of each of external.R reader function to the relevant downstream format functions as a (non-exhaustive) test that the readers are getting things into the expected format.

Reply:
That all makes sense; thanks